### ISSUE TITLE
Calibration and evaluation data - Specifics of running ESDMs within a subset of species' range

### ISSUE THREAD

**Issue by sistrurus621** [biomodhub/biomod2#593]
**Context and question**
I am modeling SDMs for three species (individually), within a subset of their range. My model training/calibration datasets were compiled from natural resource databases, and my evaluation datasets were independently collected. The independent dataset covers areas not open to the public across the study region.
I switched to biomod2 from ssdm due to the wide range of capabilities (most of all, the ability to use my own independently collected dataset for model evaluation), but I am still not what I'd consider a 'modeling expert' so bear with me.
My complete occurrence datasets (both calibration & evaluation) are relatively small (n<150).
My study area is roughly 13,500 km^2.
I am working on one species at a time, so keep that in mind when referring to the questions/code.

I need clarification on two main topics:
**1. Pseudoabsence Selection** - I have used both the 'random' and 'disk' methods. I selected 3 sets of random points using different ratios of presence:pseudoabsences. This method resulted in PAs clustered to one half of the study area, although my preliminary evaluation metrics for the models run didn't send up too many red flags. I have since switched to the 'disk' method, with a minimum distance of 10 km. 
**My question regarding PA selection**: Is there a straightforward way to scale the number of PA selections to the study area extent? I am currently using 3 different sets of PA selection for both my calibration and my evaluation data. I understand that the number of PAs:presence points will impact model performances, especially in my evaluation/test dataset, since it is much lower.

**2. Cross validation** - If I provide an independent dataset for evaluation, do I still need to use a cross validation method? If so, what would be recommended? 

In the event none of that is easy to understand, I provided example code of my PA selection methods, and data formatting below. Any suggestions, insight, and pointers is appreciated from more seasoned modelers/biomod2 users.
TIA!

##For my number of absences, I started at 1000 (due to smaller study area) and
##scaled down based on the numbers of presences
##Minimum number of absences was determined by the intermediate ratio values
##of presences:absences outlined above
##These median values performed well in preliminary modeling without overfitting

##min distance = 10km
#no max distance
myBiomodTrainData.disk.PA1 <- BIOMOD_FormatingData(resp.var = train.PA.vect, 
                                                   expl.var = env_filled,
                                                   resp.xy = train_XY, 
                                                   resp.name = myRespName, 
                                                   PA.nb.rep = 4, PA.nb.absences = 1000, 
                                                   PA.strategy = 'disk', 
                                                   PA.dist.min = 0.09)

myBiomodTrainData.disk.PA2 <- BIOMOD_FormatingData(resp.var = train.PA.vect, 
                                                   expl.var = env_filled,
                                                   resp.xy = train_XY, 
                                                   resp.name = myRespName, 
                                                   PA.nb.rep = 4, PA.nb.absences = 500, 
                                                   PA.strategy = 'disk', 
                                                   PA.dist.min = 0.09)

myBiomodTrainData.disk.PA3 <- BIOMOD_FormatingData(resp.var = train.PA.vect, 
                                                   expl.var = env_filled,
                                                   resp.xy = train_XY, 
                                                   resp.name = myRespName, 
                                                   PA.nb.rep = 4, PA.nb.absences = 350, 
                                                   PA.strategy = 'disk', 
                                                   PA.dist.min = 0.09)

##TEST DATA n=27
##min distance = 10km
#no max distance
myBiomodTestData.disk.PA1 <- BIOMOD_FormatingData(resp.var = test.PA.vect, 
                                                   expl.var = env_filled,
                                                   resp.xy = test_XY, 
                                                   resp.name = myRespName, 
                                                   PA.nb.rep = 4, PA.nb.absences = 1000, 
                                                   PA.strategy = 'disk', 
                                                   PA.dist.min = 0.09)

myBiomodTestData.disk.PA2 <- BIOMOD_FormatingData(resp.var = test.PA.vect, 
                                                   expl.var = env_filled,
                                                   resp.xy = test_XY, 
                                                   resp.name = myRespName, 
                                                   PA.nb.rep = 4, PA.nb.absences = 500, 
                                                   PA.strategy = 'disk', 
                                                   PA.dist.min = 0.09)

myBiomodTestData.disk.PA3 <- BIOMOD_FormatingData(resp.var = test.PA.vect, 
                                                  expl.var = env_filled,
                                                  resp.xy = test_XY, 
                                                  resp.name = myRespName, 
                                                  PA.nb.rep = 4, PA.nb.absences = 150, 
                                                  PA.strategy = 'disk', 
                                                  PA.dist.min = 0.09)

##PREVIOUS MODEL ATTEMPT USING 'RANDOM' PA SELECTION DATA
##FORMATTING FOR MODEL RUN ----
biomod1_1 <- BIOMOD_FormatingData(resp.var = train_resp1,
                                expl.var = env_filled,
                                resp.xy = train_xy1,
                                resp.name = 'DRME', 
                                eval.resp.var = test_resp1,
                                eval.expl.var = env_filled,
                                eval.resp.xy = test_xy1)  
##MODEL RUN - PA RATIO 1:1 ----
modelout1_1 <- BIOMOD_Modeling(bm.format = biomod1_1, 
                               modeling.id = 'AllModels', 
                               models = c('RF', 'GLM', 'CTA', 'GAM', 'GBM', 'MARS', 'XGBOOST', 'FDA'),
                               CV.strategy = 'random',
                               CV.nb.rep = 2,
                               CV.perc = 0.8,
                               OPT.strategy = 'bigboss',
                               metric.eval = c('TSS','ROC', 'KAPPA'),
                               var.import = 3, 
                               seed.val = 3763)

**Comment by HeleneBlt** [comment]
Hello there! 

Thank you for posting, and taking the time to explain the situation! 

**1. Pseudo absences** 

So first, let's have a look at the code. You can ask several sets of PA with different numbers. It will look like this : 
```
myBiomodTrainData.disk <- BIOMOD_FormatingData(resp.var = train.PA.vect,
                                               expl.var = env_filled,
                                               resp.xy = train_XY,
                                               resp.name = myRespName,
                                               PA.nb.rep = 12, 
                                               PA.nb.absences = c(rep(1000, 4),
                                                                  rep(500, 4),
                                                                  rep(350, 4)),
                                               PA.strategy = 'disk',
                                               PA.dist.min = 0.09)

```
So, you will run the models for all the PA datasets at the same time, and you can compare them more easily. 
For creating PA for evaluation dataset, I will advise using [bm_PseudoAbsences](https://biomodhub.github.io/biomod2/reference/bm_PseudoAbsences.html). Keep in mind that the use of PA implies some biases, especially PA for evaluation data. The selection depends a lot on the sampling method and effort. 

What do you mean by "_This method resulted in PAs clustered to one-half of the study area_" ? Normally, the PAs are distributed on the whole environment you provide. 
As the number of PAs, you can have a look at our vignette [PseudoAbsences](https://biomodhub.github.io/biomod2/articles/vignette_pseudoAbsences.html) 

**2. Cross validation** 
Even if you have an evaluation dataset, I will advise using cross-validation anyways. Especially because your evaluation data presents different characteristics ("_The independent dataset covers areas not open to the public across the study region._") 
For this, you can have a look at our vignette [cross-validation](https://biomodhub.github.io/biomod2/articles/vignette_crossValidation.html) and the associated papers presenting the different methods. 

Hope it helps, 
Hélène

**Comment by sistrurus621** [comment]
[.Train.1_5_PA.pdf](https://github.com/user-attachments/files/19091389/default.Train.1_5_PA.pdf)
[.Train.1_10_PA.pdf](https://github.com/user-attachments/files/19091388/default.Train.1_10_PA.pdf)

[PA.500_MINDIST.10km_TRAIN_03032025.pdf](https://github.com/user-attachments/files/19091429/PA.500_MINDIST.10km_TRAIN_03032025.pdf)

I attached two files of the 'random' method PA selections using two different ratios, and I attached what one of my 'disk' outputs looks like.


Thank you for taking the time to respond and help me with this; I am following along with the biomod2 companion document and vignettes, but having some clarification for my unique situation is very helpful!

**Comment by HeleneBlt** [comment]
Hello, 

Indeed, there is clearly a problem with the first two cases. 
Could you share : 
- the code you use to obtain it 
- the summary of your resp.var
- the summary of your expl.var 

Thanks a lot!
Hélène
